src/app/cards/cards.component.ts

```diff
@@ -212,6 +215,12 @@ export class CardsComponent implements OnInit, OnDestroy {
       );
     });
 
+    this._piaService.availableAdmins.forEach(admin => {
```

Comment:
if you want to do this, do it in the then block above as the block below is asyncrounous, so at this line you're not sure that the availableAdmins array is already populated

Reply:
availableAdmins array is populated on NgOnInit(). I guess this is the first thing which is done after the component is created. 
I tried the thing you suggested and placed this function inside them just after the availableAdmins is populated put the result at the end is NOT expected.

Reply:
In addition, the same rules apply for the previous part of this block, where we return cards for the involved users only.

Reply:
OK